llvm/lib/Transforms/Vectorize/LoopVectorizationPlanner.h

```diff
@@ -265,6 +265,12 @@ class VPBuilder {
         FPBinOp ? FPBinOp->getFastMathFlags() : FastMathFlags()));
   }
 
+  VPInstruction *createStepVector(Type *Ty) {
```

Comment:
Could introducing step vector and updating VpWidenIntOrFpInductionRecipe to not generate it itself be split off separately?

Reply:
I've opened #129508 for this, thanks